kernel/enter.c

```diff
@@ -867,6 +867,10 @@ int myst_enter_kernel(myst_kernel_args_t* args)
             myst_assume(thread->group_prev == NULL);
             while (thread->group_next)
             {
+                // We are currently only killing the thread group in
+                // SYS_exit_group. For threads aborted via a signal this flow is
+                // not hit. So another attempt to kill thread group.
+                myst_kill_thread_group();
```

Comment:
Can we do this in the signal handler before jumping back?

Reply:
@paulcallen mentioned hes looking into a solution similar to that. So I'll probably back out this particular change.